EventKit/eventkit-cloud
# Add redux toolkit

# Description of Improvement

Adds redux toolkit and replaces users actions + reducers with users slice defined in redux toolkit.
Updates the store to be created with redux toolkit, and to work with our custom middleware + enhancers.
Updates test utils to create test store with redux toolkit.

## How to test

All existing functionality and tests should work as expected.
Verify export works
Verify actions around users and groups works
Verify all tests pass

## Quality Assurance

The following items have been updated:


## Screenshots

:smile:
